### ISSUE TITLE
Feature request/issue EP>S S>EP

### ISSUE THREAD

**Issue by Th3MadHatter** [SickGear/SickGear#1155]
Hey jack,

I know you're not a fan of season searches but I keep running into temporary ban problems searching for older shows with multiple older seasons.
For example there is a hit limit (until temp ban) for 100 hits, someone wants season 5-8, each season is 25 episodes, the source used deletes single episodes after a bundle has been created (aka a season), it hits the source 26 times (25 times for EP, 1 time for season) and finds what it wants on try 26 (the season because there are no single EPs) everything works perfectly well and then it hits season 8, it works through the episode and "oh !&$% it reached 100" BANNED for an hour.

So my feature request is (against your personal preference) an option that you can set on each show (and while adding a new show) that says "prefer episode" or "prefer season" (EP>S S>EP).
I know you'e against it but some of us simply need that option, I have to wait hours to complete my entire search sometimes which could have been solved by changing over a 100 hits to 5 hits while season was preferred,

**Comment by JackDandy** [comment]
> I know you're not a fan of season searches but I keep running into temporary ban problems searching for older shows with multiple older seasons.

I have no issue with season searches, they just need a little care and attention to optimally use them.

I will assume as you are talking API hits that you are referring to newznab sites, please confirm if you can.

If you add a show where you want to pull old seasons, then that is when why you may temporarily enable the **Episode search mode AS season packs only**  for whatever provider you choose. 

If you want to _prefer_ season first and also check eps after, then set the provider **Episode search fallback**. 

To move this logic into show level is far from a trivial change, but will remain as a feature request in case others agree to vote a need for this.

**Comment by Th3MadHatter** [comment]
BTN API hits to be specific. The issue is that it searches for eps, while I personally prefer seasons (especially in older things that aren't censored after being aired again on some station or  not in love with the FCC) instead of episodes in general I know you don't and it works great for episodes (unless you're looking for pokemon or something, that has been impossible since we where both still talking to Echelon lol) but seasons can cause an issue if episodes are preferred to seasons and it can't be changed.

The problem with the "add show" thing is that (I've had this issue for at least a year but I complain enough about things so I just didn't) is that it doesn't SAVE my settings. It has some really old custom settings in place I did ages ago but, for example, I change it to HD1080 and then save, it says saved but the next time I add a new show (which is very often in my case) I get the same Custom settings I had before.

I was talking about a show level request though, the above is just a minor issue I encountered and a summary (mainly) so if others do agree it would be nice to see, even though it is a lot of work so I understand why you leave it open for votes :)

**Comment by JackDandy** [comment]
> even though it is a lot of work so I understand why you leave it open for votes

Not only that, the issue only applies to the single BTN provider, the only p2p provider with API limits, also the issue does not apply to usenet search either as that works differently.

**Comment by JackDandy** [comment]
> The problem with the "add show" thing is that (I've had this issue for at least a year but I complain enough about things so I just didn't) is that it doesn't SAVE my settings. It has some really old custom settings in place I did ages ago but, for example, I change it to HD1080 and then save, it says saved but the next time I add a new show (which is very often in my case) I get the same Custom settings I had before.

No, this is user misinterpretation. Settings are indeed saved.

Qualities without indentation are saved as their base name, that is, SD, HD , UHD, and ANY.

Indented qualities are custom, these are saved, and if you click add again (to simulate what happens after a save on the next add show) then you will see that custom is no selected with a Wanted, and Upgrade selection that you can fine tune further and save. Therefore, you can set hd1080, save, go back and then unselect the default 1080p upgrade bluray option, save, and when you go back it will be vanilla 1080p  wanted. This is as designed.

**Comment by JackDandy** [comment]
> The problem with the "add show" thing is that (I've had this issue for at least a year but I complain enough about things so I just didn't) is that it doesn't SAVE my settings. It has some really old custom settings in place I did ages ago but, for example, I change it to HD1080 and then save, it says saved but the next time I add a new show (which is very often in my case) I get the same Custom settings I had before.

This sounds like user misinterpretation, settings are saved correctly and as designed.

- Qualities listed without indentation are saved as their base name, those are; SD, HD , UHD, and ANY
- Indented qualities are *_initially_* "Custom" after "Save as defaults"

If you click "Add show" again without actually _adding_ then you will see that "Custom" is now displayed with a "smart" Wanted and Upgrade selection depending on the previously selected HD level of either 720 or 1080 -- now, you can fine tune further before hitting again "Save as defaults"...

Therefore, select "HD1080", click "Save as defaults", then menu "Add Show" (+click finally), then unselect the default Upgrade 1080p BluRay option, click "Save as defaults", and when you go back "HD1080p" and not "Custom" will be displayed. This is working as designed.

**Comment by Th3MadHatter** [comment]
Ok, let me put it really easy on that one, there is a big button saying "save as default" when adding a show.

I go and add a show, I get "custom", I change it to HD, click "Save as default" it says "saving settings as default" (or something like that) in my SG console window (I always have console open because I like console, most people find it annoying but I love seeing what happens) I add another show and it's custom again! 
So that means that the "save as default" option is either useless, not working or having trouble writing to my config file, the last one seems the most obvious but there is no error output at all.

It's fine if i'm doing it wrong, I can just edit it somewhere else if needed and I honestly don't care that much clicking an extra button, but the "save as default" button makes it the obvious choice to click for people and if it doesn't work "it's broken" (I'm not saying it is but most less techy people would say so lol).

**Comment by JackDandy** [comment]
The steps you took are not the steps listed.
1) Go to "Add show"
2) Go to step "finally"
3) Select "HD1080"
4) Click "Save as defaults"
5) Go to "Add Show"
6) Go to step "finally"
7) Unselect the default Upgrade "1080p BluRay" option
8) Click "Save as defaults"
9) Go to "Add show"
10) Go to step "finally" and observe "HD1080p" will be displayed

**Comment by JackDandy** [comment]
The steps you took are not the steps listed.
1) Go to "Add show"
2) Go to step "finally"
3) Select "HD1080"
4) Click ["Save as defaults"]
5) Go to "Add Show"
6) Go to step "finally" and observe "custom" is displayed with default Wanted and Upgrade options
7) Unselect the default Upgrade "1080p BluRay" option
8) Click ["Save as defaults"]
9) Go to "Add show"
10) Go to step "finally" and observe "HD1080p" will be displayed

**Comment by JackDandy** [comment]
The steps you took are not quite the entire steps listed...
Your step 7 is the step 5 listed below...
1) Go to "Add show", then step "finally"
2) Select "HD1080"
3) Click ["Save as defaults"]
4) Use browser refresh to action a... Go to "Add Show"
5) Go to step "finally" and observe "Custom" is displayed with default options for Wanted and Upgrade 
6) Click the default Upgrade ["1080p BluRay"] option to *unselect* it and its magnify glass will disappear
7) Click ["Save as defaults"]
8) Use browser refresh to action a... Go to "Add Show"
9) Go to step "finally" and observe "HD1080p" will be displayed

**Comment by Th3MadHatter** [comment]
Haha that worked!

It's obvious now that you say it like this, Custom>Profile so once the Custom profile meets one of the "configured profiles" it will switch to that.
For me that makes sense, for less tech savvy people it won't however, they will do the same thing I did because of simplicity in other programs, you select what you want and hit "save as default" but that doesn't work, you have to deselect things to force it into one of the profiles (which, again, makes sense!).
Maybe it might be an idea to set a simple option where Profile>Custom if saved as default. 
I've seen this "issue" since SickRage (and it's been a long time since I used that lol) but just ignored it because clicking a couple extra buttons isn't that big of a deal, even for power users, but I'm pretty sure there are a lot of people that just think it's broken because they don't understand the way it's set up to work and ignore the issue because it's a minor inconvenience.

So as a very very minor feature request, if you find the time and feel like it, add a single line rule where saving a default profile Trumps any custom setting, if people want to change it to an upgrade they'll quickly switch back to custom (or another profile) but I think it would give the save option a more "user friendly feeling".
That's just my POV of course, like I said I thought I just screwed up one of my settings and broke it because I tend to tinker with stuff haha.

But hey, thanks! I now have the default profile I've wanted for ages and the setup made complete sense (to you and me) :-)

Back to the topic at hand because I hijacked my own post discussing something completely different lol.

**Comment by Th3MadHatter** [comment]
You don't have to summarize it I wasn't attacking you, the program, or service you provide to the public, AT ALL.
If the summarization was for other readers I completely understand, SG does not break any form of ToS in any way within my current knowledge of both services used by the program or services made available by it's hosting service.

I know I've been out of the game a while but as far as I know AB, BTN, HDb and even Baka have a API hit limit (it could have changed! I haven't checked to verify because I, as a normal user, don't spam services unless I'm requested to formally or it just happens to happen during searches).

If a limiter is added, as you suggested, for certain providers of usenet or bittorrent that clearly provide legal content (which I would then suggest to take note of in the wiki, that per request a limiter could be placed by a providers query) .

My example of DDoS was, simply put, an example to why they use a limiter. One person using SickGear, or any other sickbeard fork for that matter, could never preform a simple DoS unless intentionally coded into the software, which is NOT the case in SG.

Ok after clearing out the legal reasons that might have caused you to summarize and the reason while I added clear explanation of the current state of SG, it's use and the clear statement that misuse, accidental break of general code of conduct or any other form is NOT present in the current or previous SG builds.

Adding a limiter for BTN would be great since they tend to backup show episodes and after the full season is backed up, remove the single episodes and bundle it to a complete season pack it would be a great solution.
Of course there will still be issues with TVdB seasons that are broken up into multiple named and numbered seasons meaning Season 15 is season 8-10 in real numbering or similar things (e.g. Pokemon which seems to be impossible to regex or History's Ancient Aliens) but other then that having multiple clear tries and inability to hit something, then switching to season per default and again failing to hit should help the strain it might cause on multiple users using ANY API while the software (in this case SG) already knows it's not going to get any hits because e.g. if ep 90-80 and 5-1 from something that aired in 1990 isn't found AND S1 (which contains ep 1-90) isn't found it clearly states it's not there after 16 hits instead of 91, this, of course, could be reduced even more but would be excellent for user input if it meets certain requirements (e.g. SHOW SEASON 15 = aired allow search, SHOW SEASON 15 = not aired disallow search).

**Comment by JackDandy** [comment]
I listed points because when I have said stuff in writing, due to language differences or wotnot, you have proven to miss certain things that I say, so I was just trying for an improved readability.

Anyway, like I said, some logic that may reduce hits for BTN may prove useful **if** it works with existing logic.
The further examples you talk about will not be investigated at this time, because like I explained in my very first post, seasons can be _basic_ searched with the existing available options. Or at worst, manually downloading.
